presto-memsql/src/test/java/io/prestosql/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -0,0 +1,341 @@
+                .addRoundTrip(decimalDataType(3, 1), new BigDecimal("-10.1"))
+                .addRoundTrip(decimalDataType(4, 2), new BigDecimal("2"))
+                .addRoundTrip(decimalDataType(4, 2), new BigDecimal("2.3"))
+                .addRoundTrip(decimalDataType(24, 2), new BigDecimal("2"))
+                .addRoundTrip(decimalDataType(24, 2), new BigDecimal("2.3"))
+                .addRoundTrip(decimalDataType(24, 2), new BigDecimal("123456789.3"))
+                .addRoundTrip(decimalDataType(24, 4), new BigDecimal("12345678901234567890.31"))
+                .addRoundTrip(decimalDataType(30, 5), new BigDecimal("3141592653589793238462643.38327"))
+                .addRoundTrip(decimalDataType(30, 5), new BigDecimal("-3141592653589793238462643.38327"))
+                .addRoundTrip(decimalDataType(38, 0), new BigDecimal("27182818284590452353602874713526624977"))
+                .addRoundTrip(decimalDataType(38, 0), new BigDecimal("-27182818284590452353602874713526624977"));
+    }
+
+    @Test
+    public void testDecimalExceedingPrecisionMax()
+    {
+        testUnsupportedDataType("decimal(50,0)");
+    }
+
+    @Test
+    public void testDecimalExceedingPrecisionMaxWithExceedingIntegerValues()
+    {
+        // TODO determine whether we need decimal_rounding_mode session property
+        throw new SkipException("TODO");
+    }
+
+    @Test
+    public void testDecimalExceedingPrecisionMaxWithNonExceedingIntegerValues()
+    {
+        // TODO determine whether we need decimal_rounding_mode session property
+        throw new SkipException("TODO");
+    }
+
+    @Test
+    public void testDecimalExceedingPrecisionMaxWithSupportedValues()
+    {
+        // TODO determine whether we need decimal_rounding_mode session property
+        throw new SkipException("TODO");
+    }
+
+    @Test
+    public void testPrestoCreatedParameterizedChar()
+    {
+        memSqlCharTypeTest()
+                .execute(getQueryRunner(), prestoCreateAsSelect("memsql_test_parameterized_char"));
+    }
+
+    @Test
+    public void testMemSqlCreatedParameterizedChar()
+    {
+        memSqlCharTypeTest()
+                .execute(getQueryRunner(), memSqlCreateAndInsert("tpch.memsql_test_parameterized_char"));
+    }
+
+    private DataTypeTest memSqlCharTypeTest()
+    {
+        return DataTypeTest.create()
+                .addRoundTrip(charDataType("char", 1), "")
+                .addRoundTrip(charDataType("char", 1), "a")
+                .addRoundTrip(charDataType(1), "")
+                .addRoundTrip(charDataType(1), "a")
+                .addRoundTrip(charDataType(8), "abc")
+                .addRoundTrip(charDataType(8), "12345678")
+                .addRoundTrip(charDataType(255), "a".repeat(255));
+    }
+
+    @Test
+    public void testMemSqlCreatedParameterizedCharUnicode()
+    {
+        DataTypeTest.create()
+                .addRoundTrip(charDataType(1, CHARACTER_SET_UTF8), "\u653b")
```

Comment:
can you name the String constants?

Reply:
uch... this is copied from mysql and i dont know what this constant supposed to be special about